streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -71,6 +71,48 @@
     private static final Logger log = LoggerFactory.getLogger(StreamThread.class);
     private static final AtomicInteger STREAM_THREAD_ID_SEQUENCE = new AtomicInteger(1);
 
+    /**
+     * Stream thread states are the possible state that a stream thread can be in.
+     * A thread should only be in one state at a time
+     * The expected state transition with the following defined states is:
+     *
+     *                +-----------+
+     *                |Not Running|
+     *                +-----+-----+
+     *                      |
+     *                      v
+     *                +-----+-----+
+     *          +---->| Running   |<------------+
+     *          |     +-----+-----+             |
+     *          |           |                   |
+     *          |           v                   |
+     *          |     +-----+------------+      |
+     *          <---- |Partitions        |      |
```

Comment:
So `Partitions Revoked` may transition to either of `Running`, `Partitions Assigned`, or `Pending Shutdown`, or also to any combinations thereof?